vault/resource_okta_auth_backend_test.go

```diff
@@ -0,0 +1,255 @@
+	if err != nil {
+		return err
+	}
+
+	userList, err := client.Logical().List("/auth/okta/users")
+	if err != nil {
+		return fmt.Errorf("error reading back configuration: %s", err)
+	}
+
+	if len(userList.Data["keys"].([]interface{})) != 1 {
+		return fmt.Errorf("Unexpected users present: %v", userList.Data)
+	}
+
+	user, err := client.Logical().Read("/auth/okta/users/foo")
+	if err != nil {
+		return fmt.Errorf("error reading back configuration: %s", err)
+	}
+	err = assertArrayContains([]string{"dummy"}, toStringArray(user.Data["groups"].([]interface{})))
+	if err != nil {
+		return err
+	}
+	err = assertArrayContains([]string{""}, toStringArray(user.Data["policies"].([]interface{})))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var updatedOktaAuthConfig = `
+resource "vault_okta_auth_backend" "test" {
+    description = "Testing the Terraform okta auth backend"
+    organization = "example"
+    token = "this must be kept secret"
+    group {
+        group_name = "example"
+        policies = ["three", "four"]
+    }
+    user {
+        username = "bar"
+        groups = ["example"]
+    }
+}
+`
+
+func testOktaAuthBackend_UpdatedCheck(s *terraform.State) error {
+	client := testProvider.Meta().(*api.Client)
+
+	groupList, err := client.Logical().List("/auth/okta/groups")
+	if err != nil {
+		return fmt.Errorf("error reading back configuration: %s", err)
+	}
+
+	if len(groupList.Data["keys"].([]interface{})) != 1 {
+		return fmt.Errorf("Unexpected groups present: %v", groupList.Data)
+	}
+
+	dummyGroup, err := client.Logical().Read("/auth/okta/groups/example")
+	if err != nil {
+		return fmt.Errorf("error reading back configuration: %s", err)
+	}
+	err = assertArrayContains([]string{"three", "four", "default"}, toStringArray(dummyGroup.Data["policies"].([]interface{})))
+	if err != nil {
+		return err
+	}
+
+	userList, err := client.Logical().List("/auth/okta/users")
+	if err != nil {
+		return fmt.Errorf("error reading back configuration: %s", err)
+	}
+
+	if len(userList.Data["keys"].([]interface{})) != 1 {
+		return fmt.Errorf("Unexpected users present: %v", userList.Data)
+	}
+
+	user, err := client.Logical().Read("/auth/okta/users/bar")
+	if err != nil {
+		return fmt.Errorf("error reading back configuration: %s", err)
+	}
+	err = assertArrayContains([]string{"example"}, toStringArray(user.Data["groups"].([]interface{})))
+	if err != nil {
+		return err
+	}
+
+	err = assertArrayContains([]string{""}, toStringArray(user.Data["policies"].([]interface{})))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func testOktaAuthBackend_Destroyed(state *terraform.State) error {
+	client := testProvider.Meta().(*api.Client)
+
+	authMounts, err := client.Sys().ListAuth()
+	if err != nil {
+		return err
+	}
+
+	if _, ok := authMounts["okta/"]; ok {
+		return fmt.Errorf("Auth mount not destroyed")
+	}
+
+	return nil
+}
+
+func assertEquals(expected, actual interface{}) error {
```

Comment:
I'm not super crazy about the assert* helpers; first because the Terraform style tends to just repeat the error checking and formatting, second because the way it's implemented right now makes it a bit tricky to understand which field didn't match if there's not a match, and finally because it obscures the implementation detail (that it's doing a `!=` instead of (e.g., for `time.Time`) using the built-in equality method.